packages/fxa-auth-server/lib/routes/sms.js

```diff
@@ -49,79 +49,51 @@ module.exports = (log, db, config, customs, sms) => {
+        const templateName = TEMPLATE_NAMES.get(messageId);
         const acceptLanguage = request.app.acceptLanguage;
 
-        let phoneNumberUtil, parsedPhoneNumber;
+        await customs.check(request, sessionToken.email, 'connectDeviceSms');
 
-        return customs
-          .check(request, sessionToken.email, 'connectDeviceSms')
-          .then(parsePhoneNumber)
-          .then(validatePhoneNumber)
-          .then(validateRegion)
-          .then(createSigninCode)
-          .then(sendMessage)
-          .then(logSuccess)
-          .then(createResponse);
-
-        function parsePhoneNumber() {
-          try {
-            phoneNumberUtil = PhoneNumberUtil.getInstance();
-            parsedPhoneNumber = phoneNumberUtil.parse(phoneNumber);
-          } catch (err) {
-            throw error.invalidPhoneNumber();
-          }
+        let phoneNumberUtil, parsedPhoneNumber;
+        try {
+          phoneNumberUtil = PhoneNumberUtil.getInstance();
+          parsedPhoneNumber = phoneNumberUtil.parse(phoneNumber);
+        } catch (err) {
+          throw error.invalidPhoneNumber();
         }
 
-        function validatePhoneNumber() {
-          if (!phoneNumberUtil.isValidNumber(parsedPhoneNumber)) {
-            throw error.invalidPhoneNumber();
-          }
+        if (!phoneNumberUtil.isValidNumber(parsedPhoneNumber)) {
+          throw error.invalidPhoneNumber();
         }
 
-        function validateRegion() {
-          const region = phoneNumberUtil.getRegionCodeForNumber(
-            parsedPhoneNumber
-          );
-          request.emitMetricsEvent(`sms.region.${region}`);
+        const region = phoneNumberUtil.getRegionCodeForNumber(
+          parsedPhoneNumber
+        );
+        await request.emitMetricsEvent(`sms.region.${region}`);
 
-          if (!REGIONS.has(region)) {
-            throw error.invalidRegion(region);
-          }
+        if (!REGIONS.has(region)) {
+          throw error.invalidRegion(region);
         }
 
-        function createSigninCode() {
-          if (request.app.features.has('signinCodes')) {
-            return request
-              .gatherMetricsContext({})
-              .then(metricsContext =>
-                db.createSigninCode(sessionToken.uid, metricsContext.flow_id)
-              );
-          }
-        }
-
-        function sendMessage(signinCode) {
-          return sms.send(
-            phoneNumber,
-            templateName,
-            acceptLanguage,
-            signinCode
+        let signinCode;
+        if (request.app.features.has('signinCodes')) {
+          const metricsContext = await request.gatherMetricsContext({});
+          signinCode = await db.createSigninCode(
+            sessionToken.uid,
+            metricsContext.flow_id
           );
         }
 
-        function logSuccess() {
-          return request.emitMetricsEvent(`sms.${templateName}.sent`);
-        }
+        await sms.send(phoneNumber, templateName, acceptLanguage, signinCode);
```

Comment:
What happens if `request.app.features.has('signinCodes')` returns `false`, there won't be a signin code to send.

Reply:
We have two links in config, one to use when there is a `signinCode` and one when there isn't it. So the SMS gets sent in both cases:

https://github.com/mozilla/fxa/blob/c27c02379581873b02d61255123f36805ee0ec1f/packages/fxa-auth-server/lib/senders/sms.js#L159-L166

Reply:
Thanks for that info. Do we ever use that non-signinCode SMS anymore? Should a follow on be opened to remove it if not?

Reply:
Great question! I'll have to do some research to find out the answer though.